datafusion/optimizer/tests/integration-test.rs

```diff
@@ -151,8 +151,9 @@ fn where_exists_distinct() -> Result<()> {
     let plan = test_sql(sql)?;
     let expected = "LeftSemi Join: test.col_int32 = t2.col_int32\
     \n  TableScan: test projection=[col_int32]\
-    \n  SubqueryAlias: t2\
-    \n    TableScan: test projection=[col_int32]";
+    \n  Aggregate: groupBy=[[t2.col_int32]], aggr=[[]]\
+    \n    SubqueryAlias: t2\
+    \n      TableScan: test projection=[col_int32]";
```

Comment:
I'm not sure whether this change must be good.

Because Agg and LeftSemi both can dedup, there will be a new operator, it may cause performance loss.

how do you think? @mingmwang

Reply:
Most of the time adding `Distinct`/`Aggregate` is better, it can reduce the data volumes before the `Join`(reduce the shuffle data volumes before the `Join` in a distribute engine).
Regarding the positions of `Join`/`Aggregation`, this is a another topic, in some engines they leverage  CBO rules to decide pull down/pull up aggregations through Joins.   
Snowflake implements a feature called Adaptive Aggregate placement, the basic idea is if the `Aggregate` added by the optimizer can not dedup, just skip the `Aggregate`.